Source/HTTP/httpcall.cpp

```diff
@@ -76,6 +77,40 @@ HRESULT CALLBACK HC_CALL::PerfomAsyncProvider(XAsyncOp op, XAsyncProviderData co
     {
     case XAsyncOp::Begin:
     {
+#if !HC_NOZLIB
+
+#if HC_PLATFORM == HC_PLATFORM_WIN32 || HC_PLATFORM == HC_PLATFORM_GDK
+        // Compress body before call if applicable
+        if (call->compressionAlgorithm != xbox::httpclient::HCCompressionAlgorithm::None)
```

Comment:
@jplafonta would you consider this is the right place to do body compression?

Reply:
Yes and no. I think it should be done as the first step of PerformAsync, but if you do it as part of XAsyncBegin it will run synchronously, which is likely bad.  Probably what you will need to do is queue an async task to do the compression with XTaskQueueSubmitDelayedCallback.  When that step is complete, it will continue with the first perform iteration

Reply:
The compression should also probably happen after the Fast fail check

Reply:
I just implemented these changes; can you please take a quick look at it?